test/distributed/test_c10d_nccl.py

```diff
@@ -196,36 +197,52 @@ def tearDown(self):
         torch.cuda.device_count() > 0, "GPUs are available, skipping test"
     )
     def test_init_no_gpus(self):
-        store = c10d.FileStore(self.file.name, self.world_size)
+        store = c10d.FileStore(self.file_name, self.world_size)
         with self.assertRaisesRegex(
             RuntimeError, "ProcessGroupNCCL is only supported with GPUs, no GPUs found!"
         ):
             c10d.ProcessGroupNCCL(store, self.rank, self.world_size)
 
 
-class ProcessGroupNCCLTest(TestCase):
-    MAIN_PROCESS_RANK = 0
+class ProcessGroupNCCLTest(MultiProcessTestCase):
+    def _create_process_group_nccl(self, store, rank, world_size, opts):
+        pg = c10d.ProcessGroupNCCL(store, self.rank, self.world_size, opts)
+        dist.barrier(group=pg)
+        return pg
```

Comment:
Why not just use `init_process_group`? That way you don't need to have an explicit call to `dist.barrier` since the barrier is done internally.

Reply:
discussed offline: we will keep the current testing structures (test_c10d_nccl and test_c10d_gloo testing the low-level/backend barebone apis, distributed_test.py testing the user facing APIs), ideally we want to only test the user facing APIs, created a issue to follow up: https://github.com/pytorch/pytorch/issues/68088